framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -191,6 +191,76 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
                 " is consistently oriented with regards to the blocks it neighbors";
 
     diagnosticsLog(message, _check_sidesets_orientation, flipped_pairs.size());
+
+    // Now check that there is no sideset radically flipping from one side to another
+    // We'll consider pi / 2 to be the most steep angle we'll pass
+    unsigned int num_normals_flipping = 0;
+    Real steepest_side_angles = 1;
+    for (const auto index : index_range(side_tuples))
+    {
+      if (std::get<2>(side_tuples[index]) != bid)
+        continue;
+      const auto elem_ptr = mesh->elem_ptr(std::get<0>(side_tuples[index]));
+      const auto side_id = std::get<1>(side_tuples[index]);
+
+      // Get side normal
+      const std::unique_ptr<const Elem> face = elem_ptr->build_side_ptr(side_id);
+      std::unique_ptr<FEBase> fe(FEBase::build(elem_ptr->dim(), FEType(elem_ptr->default_order())));
+      QGauss qface(elem_ptr->dim() - 1, CONSTANT);
+      fe->attach_quadrature_rule(&qface);
+      const std::vector<Point> & normals = fe->get_normals();
+      fe->reinit(elem_ptr, side_id);
+      mooseAssert(normals.size() == 1, "We expected only one normal here");
+      const auto side_normal = normals[0];
+
+      // Compare to the sideset normals of neighbor sides in that sideset
+      for (const auto neighbor : elem_ptr->neighbor_ptr_range())
+        if (neighbor)
+          for (const auto neigh_side_index : neighbor->side_index_range())
```

Comment:
Why are we looping over all neighbors? Shouldn't we just be looking at the normal pointing out from the neighbor on the current element side? If not, then I don't understand what we're doing and this needs to be made clearer somewhere

Reply:
I ll rework the explanation

we want to find the sides around the current side considered, to look at how the normal is evolving around the sideset

Reply:
comment at the top of that diagnostics code should be clearer